src/timesync/timesyncd-manager.c

```diff
@@ -1131,3 +1138,58 @@ int manager_new(Manager **ret) {
 
         return 0;
 }
+
+static int manager_save_time_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        Manager *m = userdata;
+
+        assert(m);
+
+        return manager_save_time_and_rearm(m);
+}
+
+int manager_save_time_add(Manager *m) {
+        int r;
+
+        assert(!m->event_save_time);
+
+        if (m->save_time_interval_usec != USEC_INFINITY) {
+                r = sd_event_add_time_relative(
+                                m->event, &m->event_save_time,
+                                clock_boottime_or_monotonic(),
+                                m->save_time_interval_usec,
+                                10 * USEC_PER_SEC,
+                                manager_save_time_handler, m);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add save time event: %m");
```

Comment:
so this will accmulate scheduling latencies, since the relative timer is started relative to the event loop iteration, and not the time this event source was supposed to be triggered. Maybe that's not a problem, but maybe add a comment here at least, documentint this, i.e.

```c
/* Yes, we'll accumulate scheduling latencies here, but this shouldn't matter */
```

Or alternatively, do the time calculation based on what sd_event_source_get_time() returns.

Reply:
Yeah, I don't think we care much about latencies.